Toregge/refactor posting list search context

toregge opened the pull request:
@geirst, please review

Commit:
Move BTreeKeyData class definition out to separate header file.

Commit:
Factor out portions of posting list merging from posting list search context.

Commit:
Add unit test for posting list merger.

geirst commented on searchlib/src/vespa/searchlib/btree/btree_key_data.h:
```diff
@@ -0,0 +1,87 @@
+ * Empty class to use as DataT template parameter for BTree classes to
+ * indicate that leaf nodes have no data (similar to std::set having less
+ * information than std::map).  Use of this class triggers the below
+ * partial specialization of BTreeNodeDataWrap to prevent unneeded
```
The last sentence refers to code in old location. Should be moved back to BTreeNodeDataWrap and adapted as needed.

geirst commented on searchlib/src/vespa/searchlib/attribute/posting_list_merger.h:
```diff
@@ -0,0 +1,68 @@
+    void reserveArray(uint32_t postingsCount, size_t postingsSize);
+    void allocBitVector();
+    void merge();
+    PostingVector &merge(PostingVector &v, PostingVector &temp, const StartVector &startPos) __attribute__((noinline));
```
It doesn't look like this is used publicly. Consider making it private.

Commit:
Trim down unit test.
Test adding info from multiple posting lists to bit vector.

Make internal merge method private.

Move comment.

Commit:
Adjust test names.

geirst commented:
LGTM

geirst reviewed (approved):
👍
